### DISCUSSION TITLE
Allow for normal descriptions and names to use the replace tags

### DISCUSSION THREAD

**Discussion by SquidingTin** [FakeFishGames/Barotrauma#13295] [Ideas and suggestions]
This is a modding suggestion mainly, what im referring to in the title is how talent descriptions allow for you to replace [faction] in the text xml and allows you to change the [faction] to the faction you want, allowing for 1 text to work for multiple items. While this may seem like a useless addition to the game, it wouldnt really be as you could condense a massive xml into 30 ish info parts to combine together, i will provide a snippet of what im referring to here encase you want to see how. In this code block, you can see how these short bits combining together could reduce the translation file massively, currently without this its around 240 different items that would need their own text, not even counting the amount of characters of said text. The last bit of code is how it would work on an item if it was added.

	<info.magazine_extended>[size] [firearm] magazine (Extended - [type])</info.magazine_extended>
	<info.magazine_standard>[size] [firearm] magazine ([type])</info.magazine_standard>
	<info.round>[size] Round([type])</entityname.round>

	<info.description_magazine>[mag] magazine made to hold [size] rounds. Holds up to [ammount] rounds.[info]</info.magazine>

	<info.description_noff>\n (Contains Rounds that do not collide with teammates, reducing friendly fire possibility to zero.)	<info.description_noff>

	<info.extended>Extended</info.extended>
	<info.standard>Standard</info.standard>

	<info.939_name>9x39</info.939_name>
	<info.57_name>5.7x28</info.57_name>

	<info.g14_name>G-14K Assault carbine</info.g14_name>
	<info.ak_name>AK-32MSU Perun</info.ak_name>
	
	<talentname.blueprint>Blueprint ([itemname] ammunition)</talentname.blueprint>
	<info.extender>Barrel extension([itemname])</info.extender>
	
	<info.ap>Armor-Piercing</info.ap>
	<info.rifle>Increased Powder Charge</info.rifle>

	<Description tag="info.description_magazine">
		<Replace tag="[mag]" value="info.standard" />
		<Replace tag="[size]" value="info.939_name" />
		<Replace tag="[ammount]" value="15" />
		<Replace tag="[info]" value="entitydescription.description_noff" />
	</Description>

**Comment by jigardi** [comment]
Not sure if I understand it corectly, but it sounds like this would also be useful for maintaining the game/mods in the long run.
Have my upvote!

**Reply by SquidingTin** [reply]
With talents you can make descriptions change by using the replace tag, allowing a description like 
9x39 Round(earth)
9x39 Round(water)
9x39 Round(fire)
9x39 Round(air)
to be turned into 
<info.earth>earth</info.earth>
<info.water>water</info.water>
<info.fire>fire</info.fire>
<info.air>air</info.air>

<info.939_name>9x39</info.939_name>

<info.round>[size] Round([type])</entityname.round>
meaning there doesnt need to be near duplicate text, and just have 1 universal text

(yes its kinda complicated to explain, but in short, it makes translations easier and allows for different colored text for different applications)

**Reply by jigardi** [reply]
Thanks for the explanation!
